ouroboros-network/src/Data/Sequence/Strict.hs

```diff
@@ -0,0 +1,221 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE ViewPatterns #-}
+
+-- | Strict variants of 'Seq' operations.
+--
+-- TODO: Move this to @cardano-prelude@ as a new module, @Data.Sequence.Strict@.
+module Data.Sequence.Strict
+  ( StrictSeq (.., Empty, (:<|), (:|>))
```

Comment:
I think the `..` (*) and/or the lack of a `COMPLETE` pragma (grep for it, you'll find an example in `AnchoredFragment`) are giving you the warnings about unexhaustive pattern-matches.

(*) Does the raw constructor need to be exported? I would expect not, as it could be used to convert a `Seq` with thunks as elements to a `StrictSeq` without any forcing.

Reply:
Thanks for informing me of the `COMPLETE` pragmas. I was trying to figure this out.

No, I don't think it needs to be exported. Will fix up.